solidity/contracts/libs/Call.sol

```diff
@@ -9,43 +9,78 @@ library CallLib {
         bytes data;
     }
 
-    function _multicall(address to, bytes[] memory calls) internal {
+    struct CallWithValue {
+        uint256 value;
+        Call _call;
+    }
+
+    struct CallWithCallback {
+        Call _call;
+        bytes callbackdata;
```

Comment:
Shouldn't `callbackdata` also be a `Call`?

Reply:
Can avoid defining an extra struct (and `call.call`) if you make the callback an arg, e.g.

function call(Call memory call, Call memory cb) internal returns (bytes memory) {}

Reply:
I actually did this initially
however, callbacks not made on `msg.sender` cannot be allowed and otherwise we would just duplicate `msg.sender`  across the entire array of calls

Reply:
still need to encode these

Reply:
Sorry, are you saying you can't do this?
```
Can avoid defining an extra struct (and call.call) if you make the callback an arg, e.g.

function call(Call memory call, bytes memory cb) internal returns (bytes memory) {}
```

Reply:
we can but then we have to separately encode the arrays which is suboptimal

Reply:
IMO this is cleaner and that may be worth the gas savings but I don't feel strongly

Reply:
it is cleaner but imo having the callback be a regular call that can only be on msg.sender is a footgun